### PR TITLE
UserLoginSpecial: check login token to prevent CSRF

### PR DESCRIPTION
**Pull request by macbre**
[SUS-24](https://wikia-inc.atlassian.net/browse/SUS-24)

Check login token when making a request to send "restore my password" email.

@Wikia/sustaining-team

### CODE REVIEW

File: extensions/wikia/UserLogin/UserLoginSpecialController.class.php [hunk 1]
```diff
@@ -644,6 +648,16 @@ public function mailPassword() {
 	}
 
 	/**
+	 * @see SUS-24
+	 * @throws BadRequestException
+	 */
+	public function checkWriteRequest() {
+		if ( !$this->request->wasPosted() || !hash_equals( UserLoginHelper::getLoginToken(), $this->wg->request->getVal( 'token' ) ) ) {
```
Comment (wladekb) [inline extensions/wikia/UserLogin/UserLoginSpecialController.class.php]: `Wikia\Security\Utils::matchToken` ?

Comment (macbre) [inline extensions/wikia/UserLogin/UserLoginSpecialController.class.php]: Fixed in 3cab393

**Comment by macbre** [comment]
Unit tests fixed in 4caeac9

**Comment by adamkarminski** [comment]
Ozzy approves

**Pull request merged**